python/cudf/cudf/utils/utils.py

```diff
@@ -213,6 +214,12 @@ def set_allocator(
         Enable logging (default ``False``).
         Enabling this option will introduce performance overhead.
     """
+    warnings.warn(
+        "The cudf.set_allocator function is deprecated and will be removed in "
+        "a future release. Please use rmm.reinitialize instead.",
```

Comment:
I think a link to https://docs.rapids.ai/api/rmm/stable/api.html#rmm.reinitialize is probably sufficient instead of the examples. Maybe add that link as a note in the docstring too?

Reply:
I think the link would be sufficient, maybe with a note that `allocator="managed"` translates to `managed_memory=True` since that is a change in parameter types.

Reply:
@wence- @shwina I applied my proposed suggestion.